website_data_migration/signals.py

```diff
@@ -0,0 +1,20 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from wagtail.documents.models import Document
+from .migrate_awards import migrate_data
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=Document)
+def process_xml_on_upload(sender, instance, **kwargs):
+    """Runs when a new document is uploaded or updated in Wagtail."""
+    if instance.file.name.endswith(".xml"):
```

Comment:
If we want to do this this way, perhaps we should check the whole name and not just whether it is an xml file?